docs/specs/mlnet-cli/MLNET-CLI-Specs.md

```diff
@@ -0,0 +1,969 @@
+
+- Have a CLI sample per each ML.NET sample.
+
+
+## Gated conditions for v1.0 GA (Release timeframe TBD)
+
+- Additional supported ML Tasks
+    - Recommendations
+    - Anomaly detection
+
+- Quality and benchmark conditions (TBD by the ML.NET and AutoML teams)
+
+- Azure support for exploring models and training (or vNext? - TBD)
+
+- Additional parameters in the .YAML settings file:
+    - `id_column`       // used to uniquely identify the row in dataset
+    - `groupid-column` // groups examples. Often used for Ranking
+    - `weight-column`  // indicates the weight/importance of training example. Related: `sample_weight` argument for training data such as in [Python AutoML here](https://docs.microsoft.com/en-us/python/api/azureml-train-automl/azureml.train.automl.automl?view=azure-ml-py).
+
+- Add additonal commands to do *"machine learning without code"*:
+    - *train*: It will only generate the best model .ZIP file. For example: 
+        - `mlnet train --ml-task Regression --dataset "/MyDataSets/Sales.csv"`
+    - *predict*: Having a serialized model .ZIP file, you can test a single prediction. For example:
+
+        - `mlnet predict –-model_path <path to model> –input-sample “title=foo,description=blah”`
+        - `mlnet predict –-model_path <path to model> –input-data-path “path”`
+
+- Additional scoring app templates support
+    - Console (multi-project solution)
+    - Web app (multi-project solution)
+    - Service WebAPI (multi-project solution)
+
+- Complete HTML report with model's metrics.
+
+
+## Additional features for vNext  (Release timeframe TBD)
+
+- Additional supported ML Tasks
+    - Clustering
+    - Ranking
+
+- Support for additional dataset formats in addition to text files, such as IDV or some other format.We might want to add a readertype followed by args specific to that reader.
+
+- *F#* code generation
+
+- Integration with Visual studio "Add... -> New Project"
+
+- Additional scoring app templates support
+    - Unity (multi-project solution)
+    - Desktop-WPF app (multi-project solution)
+    - Desktop-UWP app (multi-project solution)
+
+- Additional improvements
+    - Generate simplified code for additional Trainere (with no Options class in trainer) when advanced Hyper-Params are not used. This is already done for Binary-Classification and Regression.
+
+
+
+# CLI usage
+
+## Tool name
+
+**mlnet** 
+
+## Installing the tool
+
+If implemented as a [Global Tool](https://docs.microsoft.com/en-us/dotnet/core/tools/global-tools) NuGet package, you use the [dotnet tool install](dotnet-tool-install.md) .NET Core CLI command. The following example shows how to install a Global Tool in the default location:
+
+```console
+dotnet tool install -g mlnet
+```
+
+## Using the tool
+
+Once the tool is installed, you can call it by using its command. Note that the command may not be the same as the package name.
+
+You can use it with:
+
+```console
+mlnet
+```
+
+## Command 'new'
```

Comment:
can we have two commands:

new - -- only generates the project template.  Perhaps generates featurization and perhaps learning basted on heuristics, similar to the other tool's GUI wizard?
auto -- does all the automl stuff?